pkg/pool-weighted/test/foundry/CircuitBreakerLib.t.sol

```diff
@@ -0,0 +1,144 @@
+        CircuitBreakerLib.CircuitBreakerParams memory result = CircuitBreakerLib.getCircuitBreakerFields(poolState);
+
+        assertEq(result.bptPrice, bptPrice);
+        assertEq(result.weightComplement, weightComplement);
+        assertApproxEqRel(result.lowerBound, lowerBound, _MAX_RELATIVE_ERROR);
+        assertApproxEqRel(result.upperBound, upperBound, _MAX_RELATIVE_ERROR);
+
+        bytes32 initialPoolState = CircuitBreakerLib.setCircuitBreakerFields(params);
+        (uint256 initialLowerBptPriceBoundary, uint256 initialUpperBptPriceBoundary) =
+            CircuitBreakerLib.getCurrentCircuitBreakerBounds(initialPoolState, weightComplement);
+
+        uint256 expectedLowerBoundBptPrice = uint256(bptPrice).mulDown(lowerBound.powDown(weightComplement));
+        uint256 expectedUpperBoundBptPrice = uint256(bptPrice).mulDown(upperBound.powUp(weightComplement));
+
+        assertApproxEqRel(initialLowerBptPriceBoundary, expectedLowerBoundBptPrice, _MAX_RELATIVE_ERROR);
+        assertApproxEqRel(initialUpperBptPriceBoundary, expectedUpperBoundBptPrice, _MAX_RELATIVE_ERROR);
+
+        // Test that calling it with the original weightComplement retrieves exact values from the ratio cache
+        (uint256 cachedLowerBptPriceBoundary, uint256 cachedUpperBptPriceBoundary) =
+            CircuitBreakerLib.getCurrentCircuitBreakerBounds(initialPoolState, weightComplement);
+
+        assertEq(cachedLowerBptPriceBoundary, initialLowerBptPriceBoundary);
+        assertEq(cachedUpperBptPriceBoundary, initialUpperBptPriceBoundary);
+    }
+
+    function testDynamicBoundRatios(
+        uint256 initialBptPrice,
+        uint256 initialWeightComplement,
+        uint256 newWeightComplement,
+        uint256 lowerBound,
+        uint256 upperBound
+    ) public {
+        // With refBptPrice ~ 0, rounding errors make it fail
+        initialBptPrice = bound(initialBptPrice, _MIN_BPT_PRICE, _MAX_BPT_PRICE);
+        lowerBound = bound(lowerBound, _MINIMUM_BOUND_PERCENTAGE, FixedPoint.ONE);
+        upperBound = bound(upperBound, FixedPoint.ONE, _MAX_BOUND_PERCENTAGE);
+        initialWeightComplement = bound(initialWeightComplement, _MINIMUM_TOKEN_WEIGHT, _MAXIMUM_TOKEN_WEIGHT);
+        newWeightComplement = bound(newWeightComplement, _MINIMUM_BOUND_PERCENTAGE, FixedPoint.ONE);
+
+        CircuitBreakerLib.CircuitBreakerParams memory params = CircuitBreakerLib.CircuitBreakerParams({
+            bptPrice: initialBptPrice,
+            weightComplement: initialWeightComplement,
+            lowerBound: lowerBound,
+            upperBound: upperBound
+        });
+
+        // Set the initial state of the breaker
+        bytes32 initialPoolState = CircuitBreakerLib.setCircuitBreakerFields(params);
+        (uint256 lowerBptPriceBoundary, uint256 upperBptPriceBoundary) =
+            CircuitBreakerLib.getCurrentCircuitBreakerBounds(initialPoolState, newWeightComplement);
+
+        _validateWithNewComplement(
```

Comment:
I'd remove this function and just inline it - we gain nothing with this encapsulation

Reply:
It was needed for stack issues at one time; now able to inline it.